grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy

```diff
@@ -393,6 +394,12 @@ class Requisition implements Comparable<Requisition>, Serializable {
         }?.sum() ?: 0
     }
 
+    Event getMostRecentEvent() {
+        if (events?.size() > 0) {
+            return events.iterator().next()
```

Comment:
are we sure the recent event would be the first in the list?

Reply:
I think you've probably followed what is done for `Shipment`, but take a look, that there is `SortedSet` used there. 
Since `compareTo` method in `Event` domain: https://github.com/openboxes/openboxes/blob/c6d16a41c38d8b0812a98af23de80be64673ab3d/grails-app/domain/org/pih/warehouse/core/Event.groovy#L45-L54
compares by date, we are fine for the `SortedSet`, but in Requisition events are just a `Set`, so it probably won't work the way you did.